cloud/misc/rhevm.py

```diff
@@ -0,0 +1,1484 @@
+
+                # Set Delete Protection
+                del_prot = module.params.get('del_prot')
+                if r.setDeleteProtection(vminfo['name'], del_prot) is False:
+                    return RHEV_FAILED, msg
+
+                # Remove VM
+                if r.removeVM(vminfo['name']) is False:
+                    return RHEV_FAILED, msg
+                setMsg('VM has been removed.')
+                vminfo['state'] = 'DELETED'
+            else:
+                setMsg('VM was already removed.')
+            return RHEV_SUCCESS, {'changed': changed, 'msg': msg, 'vm': vminfo}
+
+    elif state == 'up' or state == 'down' or state == 'restarted':
+        name = module.params.get('name')
+        if not name:
+            setMsg("`name` is a required argument.")
+            return RHEV_FAILED, msg
+        timeout = module.params.get('timeout')
+        if r.setPower(name, state, timeout) is False:
+            return RHEV_FAILED, msg
+        vminfo = r.getVM(name)
+        return RHEV_SUCCESS, {'changed': changed, 'msg': msg, 'vm': vminfo}
+
+    elif state == 'cd':
+        name = module.params.get('name')
+        cd_drive = module.params.get('cd_drive')
+        if r.setCD(name, cd_drive) is False:
+            return RHEV_FAILED, msg
+        return RHEV_SUCCESS, {'changed': changed, 'msg': msg}
+
+
+def main():
+    global module
+    module = AnsibleModule(
+        argument_spec = dict(
+            state      = dict(default='present', choices=['ping', 'present', 'absent', 'up', 'down', 'restarted', 'cd']),
+            user       = dict(default="admin@internal"),
+            password   = dict(required=True),
+            server     = dict(default="127.0.0.1"),
+            port       = dict(default="443"),
+            ssl_verify = dict(default=True, type='bool'),
+            name       = dict(),
+            image      = dict(default=False),
+            datacenter = dict(default="Default"),
+            type       = dict(default="server", choices=['server', 'desktop', 'host']),
+            cluster    = dict(default=''),
+            vmhost     = dict(default=False),
+            vmcpu      = dict(default="2"),
+            vmmem      = dict(default="1"),
+            disks      = dict(),
+            osver      = dict(default="rhel_6x64"),
+            ifaces     = dict(aliases=['nics', 'interfaces']),
+            timeout    = dict(default=False),
+            mempol     = dict(default="1"),
+            vm_ha      = dict(default=True),
+            cpu_share  = dict(default="0"),
+            boot_order = dict(default=["network", "hd"]),
+            del_prot   = dict(default=True, type="bool"),
+            cd_drive   = dict(default=False)
+        ),
+    )
+
+    if not HAS_SDK:
+        module.fail_json(
+            msg='The `ovirtsdk` module is not importable. Check the requirements.'
+        )
+
+    rc = RHEV_SUCCESS
+    try:
+        rc, result = core(module)
+    except Exception, e:

```

Comment:
We've recently added the requirement that new modules need to be compatible with python 3. This PR hasn't been updated since that check was added to Travis. Could you please update?

python3.4 -m compileall -fq rhevm.py 
*** Error compiling 'rhevm.py'...
  File "rhevm.py", line 1471
    except Exception, e:


http://docs.ansible.com/ansible/developing_modules_python3.html